pkgs/build-support/fetchpatch/default.nix

```diff
@@ -9,15 +9,18 @@ let
   # 0.3.4 would change hashes: https://github.com/NixOS/nixpkgs/issues/25154
   patchutils = buildPackages.patchutils_0_3_3;
 in
-{ stripLen ? 0, extraPrefix ? null, excludes ? [], includes ? [], revert ? false, ... }@args:
+{ excludes ? []
+, extraPrefix ? null
+, includes ? []
+, revert ? false
+, postFetch ? ""
+, stripLen ? 0
+, ...
+}@args:
 
-let
-  # Make base-64 encoded SRI hash filename-safe using RFC 4648 §5
-  tmpname = lib.replaceStrings [ "+" "/" "=" ] [ "-" "_" "" ] args.sha256;
-in
 fetchurl ({
   postFetch = ''
-    tmpfile="$TMPDIR/${tmpname}"
+    tmpfile="$TMPDIR/patch"
```

Comment:
Nix always creates a fresh TMPDIR, regardless of whether the build is properly sandboxed. (And I suppose nix-shell may also be an exception)

Regardless, this could even use a relative path, because the working directory is also temporary.